test/tests/functional/pbs_ralter.py

```diff
@@ -1516,3 +1550,254 @@ def test_failed_ralter(self):
+                                 check_log=False)
+
+        t_duration, t_start, t_end = self.get_resv_time_info(rid)
+
+        self.assertEqual(t_end, end+10)
+        self.assertEqual(t_start, start)
+        self.assertEqual(t_duration, 30)
+
+        # Submit a job to the reservation and change its duration.
+        self.submit_job_to_resv(rid)
+
+        temp_end = t_end
+        temp_start = t_start
+
+        self.alter_a_reservation(rid, temp_start, temp_end, sequence=2,
+                                 a_duration=40, check_log=False)
+
+        t_duration, t_start, t_end = self.get_resv_time_info(rid)
+        self.assertEqual(t_end, temp_end+10)
+        self.assertEqual(t_start, temp_start)
+        self.assertEqual(t_duration, 40)
+
+    def test_adv_resv_dur_and_endtime_before_start(self):
+        """
+        Test that duration and end time of reservation can be changed together.
+        In this case the start time of reservation may also change.
+        """
+
+        offset = 20
+        duration = 20
+        shift = 10
+        rid, start, end = self.submit_and_confirm_reservation(offset, duration)
+
+        self.alter_a_reservation(rid, start, end, shift=shift,
+                                 alter_e=True, a_duration=40, check_log=False)
+
+        t_duration, t_start, t_end = self.get_resv_time_info(rid)
+
+        self.assertEqual(t_end, end+10)
+        self.assertEqual(t_start, start-10)
+        self.assertEqual(t_duration, 40)
+
+        # Submit a job to the reservation and change its start time.
+        self.submit_job_to_resv(rid)
+        temp_end = t_end
+        temp_start = t_start
+
+        self.alter_a_reservation(rid, temp_start, temp_end, shift=shift,
+                                 alter_e=True, sequence=2,
+                                 a_duration=50, check_log=False)
+        t_duration, t_start, t_end = self.get_resv_time_info(rid)
+        self.assertEqual(t_end, temp_end+10)
+        self.assertEqual(t_start, temp_start)
+        self.assertEqual(t_duration, 50)
+
+    def test_adv_resv_dur_and_starttime_before_start(self):
+        """
+        Test duration and starttime of reservation can be changed together.
+        In this case the endtime will change accordingly
+        """
+
+        offset = 60
+        duration = 20
+        shift = 10
+        rid, start, end = self.submit_and_confirm_reservation(offset, duration)
+
+        self.alter_a_reservation(rid, start, end, shift=shift,
+                                 alter_s=True, a_duration=30, check_log=False)
+
+        t_duration, t_start, t_end = self.get_resv_time_info(rid)
+
+        self.assertEqual(t_end, end+20)
+        self.assertEqual(t_start, start+10)
+        self.assertEqual(t_duration, 30)
+
+        # Submit a job to the reservation and change its start time.
+        self.submit_job_to_resv(rid)
+        temp_end = t_end
+        temp_start = t_start
```

Comment:
Why do we care if there is a job in the reservation?  The only time it matters if the reservation is empty is if it is running and you are changing the start time.

Reply:
Just an extra testing to check reservation start time can be changed when there is job in reservation that is not running.